docs/02-app/02-api-reference/02-file-conventions/middleware.mdx

```diff
@@ -0,0 +1,112 @@
+---
+title: middleware.js
+description: API reference for the middleware.js  file.
+related:
+  title: Learn more about Middleware
+  links:
+    - app/building-your-application/routing/middleware
+---
+
+The `middleware.ts` (or .js) file is used to write [Middleware](/docs/app/building-your-application/routing/middleware) and run code on the server before a request is completed. Then, based on the incoming request, you can modify the response by rewriting, redirecting, modifying the request or response headers, or responding directly.
+
+Middleware executes before routes are rendered. It's particularly useful for implementing custom server-side functionalities like authentication, logging, or handling redirects.
+
+Use the file `middleware.ts` (or .js) in the root of your project to define Middleware. For example, at the same level as `app` or `pages`, or inside `src` if applicable.
+
+```tsx filename="middleware.ts" switcher
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
+
+// This function can be marked `async` if using `await` inside
+export function middleware(request: NextRequest) {
+  return NextResponse.redirect(new URL('/home', request.url))
+}
+
+export const config = {
+  matcher: '/about/:path*',
+}
+```
+
+```js filename="middleware.js" switcher
+import { NextResponse } from 'next/server'
+
+// This function can be marked `async` if using `await` inside
+export function middleware(request) {
+  return NextResponse.redirect(new URL('/home', request.url))
+}
+
+export const config = {
+  matcher: '/about/:path*',
+}
+```
+
+## Exports
+
+### Middleware function
+
+The `middleware.ts` (or `.js`) file must export a single Middleware function, either as a default or named export. Note that multiple Middleware exports from the same file are not supported.
```

Comment:
Clarify that it can either be a default export, or a function named `middleware`

```suggestion
The file must export a single function, either as a default export or named `middleware`. Note that multiple middleware from the same file are not supported.
```